aten/src/ATen/cuda/jiterator_impl.h

```diff
@@ -65,37 +65,42 @@ int jitted_can_vectorize_up_to(const TensorIteratorBase& iter) {
   return result;
 }
 
-template<int N>
-static std::unique_ptr<OffsetCalculator<N>> make_unique_input_offset_calculator(const TensorIteratorBase& iter) {
+template<bool IS_INPUT, int N>
```

Comment:
doesn't look like IS_INPUT needs to be template arg?

Reply:
IIUC, compiler will try to optimize away the dead code path for such template arg. 

so the switch is done at compilation time, rather than run time.